### ISSUE TITLE
Pre-trained model and eval 

### ISSUE THREAD

**Issue by Tong-Gao-dream** [MarSaKi/ETPNav#18]
Respected author, thank you very much for your excellent work. I have placed the pre-trained model parameters in the path as indicated in the README. However, I received the message "This is not supported for all configurations of models and can yield errors."

1. Should I modify any specific paths?
2. When eval the model, it takes a very long time (around 3 days with one A6000 card), and after running, the path "ETPNav-main/data/logs/eval_results/release_r2r/" remains empty. Do you know how to resolve this issue?

├── pretrained
│   ├── ETP
│   │   ├── data
│   │   │   ├── datasets
│   │   │   └── logs
│   │   ├── mlm.sap_habitat_depth
│   │   │   ├── ckpts
│   │   │   └── logs
│   │   ├── mlm.sap_r2r
│   │   │   └── ckpts
│   │   ├── mlm.sap_rxr
│   │   │   └── ckpts
│   │   └── pretrained
│   │       └── ETP
│   ├── LXMERT
│   └── r2r_ce
│       └── mlm.sap_habitat_depth
│           ├── ckpts
│           └── logs
├── data
│   ├── datasets
│   │   ├── mp3d_habitat
│   │   ├── R2R_VLNCE_v1-2_preprocessed
│   │   │   ├── envdrop
│   │   │   ├── joint_train_envdrop
│   │   │   ├── test
│   │   │   ├── train
│   │   │   ├── val_seen
│   │   │   └── val_unseen
│   │   ├── R2R_VLNCE_v1-2_preprocessed_BERTidx
│   │   │   ├── test
│   │   │   ├── train
│   │   │   ├── val_seen
│   │   │   └── val_unseen
│   │   └── RxR_VLNCE_v0_enc_xlmr
│   │       ├── test_challenge
│   │       ├── train
│   │       ├── val_seen
│   │       └── val_unseen
│   ├── ddppo-models
│   ├── logs
│   │   ├── checkpoints
│   │   │   ├── release_r2r
│   │   │   └── release_rxr
│   │   ├── eval_results
│   │   │   ├── release_r2r
│   │   │   └── release_rxr
│   │   ├── running_log
│   │   └── tensorboard_dirs
│   │       └── release_r2r
│   ├── scene_datasets
│   │   └── mp3d
│   │       ├── 17DRP5sb8fy
│   │       ├── 1LXtFkjw3qL
│   │       ├── 1pXnuDYAj8r
│   │       ├── 29hnd4uzFmX
...

Initalizing the ETP model ...
model_config DEPTH_ENCODER:
  backbone: resnet50
  cnn_type: VlnResnetDepthEncoder
  ddppo_checkpoint: /data3/gaotong/python/ETPNav-main/data/ddppo-models/gibson-4plus-mp3d-train-val-test-resnet50.pth
  output_size: 256
INSTRUCTION_ENCODER:
  bidirectional: True
  dataset_vocab: data/datasets/R2R_VLNCE_v1-2_preprocessed/train/train.json.gz
  embedding_file: data/datasets/R2R_VLNCE_v1-2_preprocessed/embeddings.json.gz
  embedding_size: 50
  final_state_only: True
  fine_tune_embeddings: False
  hidden_size: 128
  rnn_type: LSTM
  sensor_uuid: instruction
  use_pretrained_embeddings: True
  vocab_size: 2504
NUM_ANGLES: 12
PROGRESS_MONITOR:
  alpha: 1.0
  use: False
RGB_ENCODER:
  cnn_type: TorchVisionResNet50
  output_size: 512
SEQ2SEQ:
  use_prev_action: False
STATE_ENCODER:
  hidden_size: 512
  rnn_type: GRU
TORCH_GPU_ID: 0
VISUAL_DIM:
  directional: 128
  vis_hidden: 768
ablate_depth: False
ablate_instruction: False
ablate_rgb: False
consume_ghost: True
fix_lang_embedding: False
fix_pano_embedding: False
merge_ghost: True
policy_name: PolicyViewSelectionETP
pretrained_path: pretrained/ETP/mlm.sap_r2r/ckpts/model_step_82500.pt
spatial_output: False
task_type: r2r
use_depth_embedding: True
use_sprels: True
You are using a model of type bert to instantiate a model of type . This is not supported for all configurations of models and can yield errors.
Some weights of the model checkpoint at None were not used when initializing GlocalTextPathNavCMT: ['bert.global_encoder.encoder.x_layers.3.lang_output.dense.weight', 'bert.global_encoder.encoder.x_layers.3.lang_output.LayerNorm.bias', 'bert.global_encoder.encoder.x_layers.2.lang_output.LayerNorm.weight', 'bert.global_encoder.encoder.x_layers.1.lang_self_att.output.dense.weight', 'bert.global_encoder.encoder.x_layers.0.lang_self_att.output.LayerNorm.weight', 'bert.global_encoder.encoder.x_layers.2.lang_self_att.output.dense.weight', 'bert.global_encoder.encoder.x_layers.2.lang_inter.dense.weight', 'bert.global_encoder.encoder.x_layers.0.lang_self_att.self.query.bias', 'bert.global_encoder.encoder.x_layers.3.lang_self_att.self.query.weight', 'bert.global_encoder.encoder.x_layers.1.lang_self_att.output.dense.bias', 'bert.global_encoder.encoder.x_layers.3.lang_self_att.self.key.bias', 'bert.global_encoder.encoder.x_layers.3.lang_self_att.self.query.bias', 'bert.global_encoder.encoder.x_layers.2.lang_self_att.self.key.weight', 'bert.global_encoder.encoder.x_layers.1.lang_self_att.self.query.bias', 'bert.global_encoder.encoder.x_layers.1.lang_output.LayerNorm.weight', 'bert.global_encoder.encoder.x_layers.2.lang_self_att.output.dense.bias', 'bert.global_encoder.encoder.x_layers.3.lang_self_att.output.LayerNorm.weight', 'mlm_head.predictions.decoder.weight', 'bert.global_encoder.encoder.x_layers.0.lang_self_att.output.dense.weight', 'bert.global_encoder.encoder.x_layers.0.lang_self_att.output.LayerNorm.bias', 'bert.global_encoder.encoder.x_layers.3.lang_self_att.self.value.bias', 'bert.global_encoder.encoder.x_layers.3.lang_self_att.output.dense.bias', 'bert.global_encoder.encoder.x_layers.3.lang_self_att.self.value.weight', 'mlm_head.predictions.bias', 'bert.global_encoder.encoder.x_layers.3.lang_self_att.output.dense.weight', 'bert.global_encoder.encoder.x_layers.1.lang_self_att.output.LayerNorm.weight', 'bert.global_encoder.encoder.x_layers.2.lang_output.dense.bias', 'mlm_head.predictions.transform.dense.bias', 'mlm_head.predictions.transform.dense.weight', 'bert.global_encoder.encoder.x_layers.0.lang_output.dense.weight', 'bert.global_encoder.encoder.x_layers.2.lang_self_att.self.query.weight', 'bert.global_encoder.encoder.x_layers.2.lang_self_att.output.LayerNorm.bias', 'mlm_head.predictions.transform.LayerNorm.bias', 'bert.global_encoder.encoder.x_layers.1.lang_inter.dense.bias', 'bert.global_encoder.encoder.x_layers.0.lang_inter.dense.weight', 'mlm_head.predictions.transform.LayerNorm.weight', 'bert.global_encoder.encoder.x_layers.2.lang_self_att.self.value.weight', 'bert.global_encoder.encoder.x_layers.2.lang_output.LayerNorm.bias', 'bert.global_encoder.encoder.x_layers.3.lang_self_att.output.LayerNorm.bias', 'bert.global_encoder.encoder.x_layers.0.lang_self_att.self.value.weight', 'bert.global_encoder.encoder.x_layers.1.lang_self_att.self.value.bias', 'bert.global_encoder.encoder.x_layers.2.lang_output.dense.weight', 'bert.global_encoder.encoder.x_layers.3.lang_self_att.self.key.weight', 'bert.global_encoder.encoder.x_layers.0.lang_output.LayerNorm.weight', 'bert.global_encoder.encoder.x_layers.1.lang_self_att.self.key.weight', 'bert.global_encoder.encoder.x_layers.0.lang_self_att.self.value.bias', 'bert.global_encoder.encoder.x_layers.0.lang_output.dense.bias', 'bert.global_encoder.encoder.x_layers.1.lang_self_att.self.query.weight', 'bert.global_encoder.encoder.x_layers.1.lang_self_att.output.LayerNorm.bias', 'bert.global_encoder.encoder.x_layers.2.lang_self_att.self.query.bias', 'bert.global_encoder.encoder.x_layers.3.lang_inter.dense.bias', 'bert.global_encoder.encoder.x_layers.0.lang_self_att.self.key.weight', 'bert.global_encoder.encoder.x_layers.1.lang_output.dense.bias', 'bert.global_encoder.encoder.x_layers.0.lang_output.LayerNorm.bias', 'bert.global_encoder.encoder.x_layers.1.lang_output.dense.weight', 'bert.global_encoder.encoder.x_layers.3.lang_output.dense.bias', 'bert.global_encoder.encoder.x_layers.2.lang_self_att.self.value.bias', 'bert.global_encoder.encoder.x_layers.2.lang_inter.dense.bias', 'bert.global_encoder.encoder.x_layers.0.lang_self_att.self.key.bias', 'bert.global_encoder.encoder.x_layers.0.lang_self_att.output.dense.bias', 'bert.global_encoder.encoder.x_layers.0.lang_self_att.self.query.weight', 'bert.global_encoder.encoder.x_layers.3.lang_inter.dense.weight', 'bert.global_encoder.encoder.x_layers.1.lang_self_att.self.key.bias', 'bert.global_encoder.encoder.x_layers.0.lang_inter.dense.bias', 'bert.global_encoder.encoder.x_layers.1.lang_self_att.self.value.weight', 'bert.global_encoder.encoder.x_layers.3.lang_output.LayerNorm.weight', 'bert.global_encoder.encoder.x_layers.1.lang_inter.dense.weight', 'bert.global_encoder.encoder.x_layers.1.lang_output.LayerNorm.bias', 'bert.global_encoder.encoder.x_layers.2.lang_self_att.self.key.bias', 'bert.global_encoder.encoder.x_layers.2.lang_self_att.output.LayerNorm.weight']
- This IS expected if you are initializing GlocalTextPathNavCMT from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).
- This IS NOT expected if you are initializing GlocalTextPathNavCMT from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).
2024-12-24 13:24:52,680 Agent parameters: 299.14 MB. Trainable: 140.79 MB.
2024-12-24 13:24:52,681 Finished setting up policy.
2024-12-24 13:24:52,682 Traning Starts... GOOD LUCK!

**Comment by Tong-Gao-dream** [comment]
The problem has been resolved and the accuracy mentioned in the paper has been reproduced.

**Comment by GalaxyS1ayer** [comment]
Can I ask how you solved this? I'm encountering the same issue—my local training results differ significantly from those in the paper, but when I validated using the weights provided by the authors, the results matched.
